src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -3997,6 +4121,133 @@ namespace ZoneTempPredictorCorrector {
+						} else {
+							InfilOAACHHM = 0.0;
+						}
+
+						Zone(ZoneNum).InfilOAAirChangeRateHM = InfilOAACHHM;
+						Zone(ZoneNum).MCPIHM = InfilOAACHHM * CpAir * AirDensity / SecInHour * Zone( ZoneNum ).Volume;
+
+					} // Hybrid model infiltration calcualtion end
+
+					// Hybrid modeling internal thermal mass calcualtion start
+					if ( HybridModelZone( ZoneNum ).InternalThermalMassCalc ){
+
+						if ( SumSysMCpT == 0 && ZT( ZoneNum ) != PreviousMeasuredZT1( ZoneNum ) && UseZoneTimeStepHistory ){ // HM calculation only when HVAC off
+							// Calculate air capacity using UseAnalyticalSolution
+							if ( TempDepCoef == 0.0 ) {
+								AirCapHM = TempIndCoef / ( ZT( ZoneNum ) - PreviousMeasuredZT1( ZoneNum ) ); // Inverse equation
+							}
+							else {
+								Real64 AirCapHM_temp = ( TempIndCoef - TempDepCoef * PreviousMeasuredZT1( ZoneNum ) ) / ( TempIndCoef - TempDepCoef * ZT( ZoneNum ) );
+								if ( (AirCapHM_temp > 0 ) && ( AirCapHM_temp != 1 ) ){ // Avoide IND
+									AirCapHM = TempDepCoef / std::log( AirCapHM_temp ); // Inverse equation
+								}
+								else {
+									AirCapHM = TempIndCoef / ( ZT( ZoneNum ) - PreviousMeasuredZT1( ZoneNum ) );
+								}
+							}
+
+							// Calculate multiplier
+							if ( abs( ZT( ZoneNum ) - PreviousMeasuredZT1( ZoneNum ) ) > 0.05 ){ // Filter
+								MultpHM = AirCapHM / ( Zone( ZoneNum ).Volume * PsyRhoAirFnPbTdbW( OutBaroPress, ZT( ZoneNum ), ZoneAirHumRat( ZoneNum ) ) * PsyCpAirFnWTdb( ZoneAirHumRat( ZoneNum ), ZT( ZoneNum ) ) ) * ( TimeStepSys * SecInHour ); // Inverse equation
+								if ( ( MultpHM < 1.0 ) || ( MultpHM > 30.0 ) ){ // Temperature capacity multiplier greater than 1 and less than 30
+									MultpHM = 1.0; // Default value 1.0 
+								}
+							}
+							else {
+								MultpHM = 1.0; // Default value 1.0 
+							}
+						}
+
+
+						// For timestep output
+						Zone(ZoneNum).ZoneVolCapMultpSensHM = MultpHM;
+						
+						// Calculate the average multiplier of the zone for the whole running period
+						{
+							// count for hybrid model calculations
+							if( MultpHM > 1.0 ){
+								Zone( ZoneNum ).ZoneVolCapMultpSensHMSum += MultpHM;
+								Zone( ZoneNum ).ZoneVolCapMultpSensHMCountSum++;
+							}
+							
+							// Calculate and store the multiplier average at the end of HM simulations 
+							if ( DayOfYear == HybridEndDayOfYear && EndDayFlag ){
+								HMMultiplierAverage = Zone( ZoneNum ).ZoneVolCapMultpSensHMSum / Zone( ZoneNum ).ZoneVolCapMultpSensHMCountSum;
+								Zone( ZoneNum ).ZoneVolCapMultpSensHMAverage = HMMultiplierAverage;
+							}
+						}
+					} // Hybrid model internal thermal mass calcualtion end
+				}
+
+				// Update zone temperatures in the previous steps
+				PreviousMeasuredZT3( ZoneNum ) = PreviousMeasuredZT2( ZoneNum );
+				PreviousMeasuredZT2( ZoneNum ) = PreviousMeasuredZT1( ZoneNum );
+				PreviousMeasuredZT1( ZoneNum ) = ZT( ZoneNum );
```

Comment:
@sanghlee-lbnl The trigger to make time step downshifted is ZoneTempChange (different between current time step and previous time step)

		if ( ZoneTempChange > MaxZoneTempDiff && ! KickOffSimulation ) { in HVACManager

It may independent of system performance.

There is no system in an example file: AirflowNetwork3zVent.idf. If I replace output variable report frequency from timestep to detailed, here is a port of output:

![capture](https://cloud.githubusercontent.com/assets/5970656/20887678/ae4e2724-bac9-11e6-831b-2af46702580a.JPG)

Therefore, even though SumSysMCpT = 0, it is possible that the time step may be down shifted.